corehq/apps/users/views/mobile/users.py

```diff
@@ -79,11 +78,12 @@ def dispatch(self, request, *args, **kwargs):
     @property
     @memoized
     def custom_data(self):
+        is_custom_data_post = self.request.method == "POST" and self.request.POST['form_type'] == "update-user"
         return CustomDataEditor(
             field_view=UserFieldsView,
             domain=self.domain,
             existing_custom_data=self.editable_user.user_data,
-            post_dict=self.request.POST if self.request.method == "POST" else None,
+            post_dict=self.request.POST if is_custom_data_post else None,
```

Comment:
this is the bugfix - when you posted from the commtrack tab it didn't have the CUD context and so was overriding the UI with a dict that had no values in it.